### DISCUSSION TITLE
Failed to bind UDP socket and unable to connect tx and rx in gnb

### DISCUSSION THREAD

**Discussion by rajugbv** [srsran/srsRAN_Project#105] [General Help]
Hi everyone ,i am trying to build standalone project using usrp B210 and usrp b200 using srsran and open5gs 

error:-
Failed to connect UDP how can i resolve it and unable to establish connect between tx and rx
i have used default USIM details and also installed webui on both the laptops insert the same values 
i dont know why it is going into underflow 

transmitter laptop 1 responds:-

raju@raju-ThinkPad-P16s-Gen-1:~/srsRAN_Project/build/apps/gnb$ sudo ./gnb -c /home/raju/srsRAN_Project/configs/gnb_rf_b200_tdd_n78_20mhz.yml
[sudo] password for raju: 
Lower PHY in dual executor mode.

--== srsRAN gNB (commit 0112729f4) ==--

Connecting to AMF on 127.0.1.100:38412
Failed to bind UDP socket to 127.0.0.1:2152. Address already in use
Available radio types: uhd.
[INFO] [UHD] linux; GNU C++ version 9.4.0; Boost_107100; UHD_4.4.0.0-177-g0e50a37b
Making USRP object with args 'type=b200,num_recv_frames=64,num_send_frames=64'
Cell pci=1, bw=20 MHz, dl_arfcn=632628 (n78), dl_freq=3489.42 MHz, dl_ssb_arfcn=632544, ul_freq=3489.42 MHz

==== gNodeB started ===
Type <t> to view trace
t
Late: 0; Underflow: 1; Overflow: 0;
Late: 0; Underflow: 2; Overflow: 0;
Late: 0; Underflow: 2; Overflow: 0;
Late: 0; Underflow: 1; Overflow: 0;
Late: 0; Underflow: 1; Overflow: 0;
Late: 0; Underflow: 2; Overflow: 0;
Late: 0; Underflow: 1; Overflow: 0;

**Comment by rajugbv** [comment]
receiver laptop responds

raju@raju-ThinkPad-P16s-Gen-1:~$ sudo srsue Downloads/ue.conf 
Reading configuration file Downloads/ue.conf...

Built in RelWithDebInfo mode using 22.10.0.

Opening 1 channels in RF device=uhd with args=clock=internal
Supported RF device list: UHD file
[INFO] [UHD] linux; GNU C++ version 9.3.0; Boost_107100; UHD_3.15.0.0-release
[INFO] [LOGGING] Fastpath logging disabled at runtime.
[INFO] [B200] Loading firmware image: /usr/share/uhd/images/usrp_b200_fw.hex...
Opening USRP channels=1, args: type=b200,master_clock_rate=23.04e6
[INFO] [UHD RF] RF UHD Generic instance constructed
[INFO] [B200] Detected Device: B200mini
[INFO] [B200] Loading FPGA image: /usr/share/uhd/images/usrp_b200mini_fpga.bin...
[INFO] [B200] Operating over USB 3.
[INFO] [B200] Initialize CODEC control...
[INFO] [B200] Initialize Radio control...
[INFO] [B200] Performing register loopback test... 
[INFO] [B200] Register loopback test passed
[INFO] [B200] Asking for clock rate 23.040000 MHz... 
[INFO] [B200] Actually got clock rate 23.040000 MHz.
Setting manual TX/RX offset to 300 samples
Waiting PHY to initialize ... done!
Attaching UE...
t
Enter t to stop trace.
--- disconnected ---
t--- disconnected ---

Enter t to restart trace.
t
Enter t to stop trace.
--- disconnected ---

**Reply by rajugbv** [reply]
![image](https://github.com/srsran/srsRAN_Project/assets/41448193/79310e8b-3aeb-410d-a09a-c3ed3d78e7d7)

**Comment by pgawlowicz** [comment]
Hi, back in June, the srsUE did not support bandwidth of 20Mhz. Now it does. Please check the newest srsRAN_4G project version (23.11) and the updated tutorial https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html
